graphql_compiler/schema_generation/schema_graph.py

```diff
@@ -88,19 +89,19 @@ def get_superclasses_from_class_definition(class_definition):
     return []
 
 
+@six.python_2_unicode_compatible
+@six.add_metaclass(ABCMeta)
 class SchemaElement(object):
-    ELEMENT_KIND_VERTEX = 'vertex'
-    ELEMENT_KIND_EDGE = 'edge'
-    ELEMENT_KIND_NON_GRAPH = 'non-graph'
 
-    def __init__(self, class_name, kind, abstract, properties, class_fields):
+    # Since an abstract class without any abstract methods can be instantiated, we make the
+    # init method abstract in order to make it impossible to instantiate this class.
+    # Inspiration came from: https://stackoverflow.com/questions/44800659/python-abstract-class.
```

Comment:
The link is broken: `https://stackoverflow.com/questions/44800659/python-abstract-class` is wrong -- compare it with the PR comment below. It currently happens to redirect to the right place, but let's not depend on that and fix it instead. Also, `.` is a valid URL character, so avoid putting `.` at the end of links.

Reply:
Fixed in "nits" pr.